### PR TITLE
Include_dir pull request, with testers and no dummyfiles in repository

### PR DESCRIPTION
**Pull request by anttiryt**
pr/151 was never merged as it was unclear whether perl would mixup file pointers with multiple config subdirs. After testing (my) perl didn't seem to mix up, so I just wrote testers to prove this all six config files get processed as should. it is easy to add more config subdirs if there are doubts. 

Tester now fully tests rsnapshot basic functionality.

### CODE REVIEW

**Comment by DrHyde** [comment]
This looks reasonable

File: t/include_conf_dir/conf/include_conf_dir.conf.in [hunk 1]
```diff
@@ -0,0 +1,9 @@
+config_version	1.2
+
+snapshot_root	@SNAP@
+cmd_rsync		@RSYNC@
+
+interval		hourly	6
+
+
```
Comment (sgpinkus) [inline t/include_conf_dir/conf/include_conf_dir.conf.in:8]: Please rm unecessary whitespace in these confs

File: t/include_conf_dir/include_conf_dir.t.in [hunk 2]
```diff
@@ -0,0 +1,84 @@
+#!@PERL@
```
Comment (sgpinkus) [inline t/include_conf_dir/include_conf_dir.t.in:1]: Test setup is quite complicated. Quite hard to follow. I'm against dynamically generating new files/folders for a given test. Should be able to use what's already there. If not just add a new static test dir or file. can even backup your own test config file.
